presto-main/src/main/java/io/prestosql/sql/planner/RelationPlanner.java

```diff
@@ -411,15 +409,15 @@ else if (firstDependencies.stream().allMatch(right::canResolve) && secondDepende
 
         if (node.getType() != INNER) {
             for (Expression complexExpression : complexJoinExpressions) {
-                Set<InPredicate> inPredicates = subqueryPlanner.collectInPredicateSubqueries(complexExpression, node);
-                if (!inPredicates.isEmpty()) {
-                    InPredicate inPredicate = Iterables.getLast(inPredicates);
-                    throw semanticException(NOT_SUPPORTED, inPredicate, "IN with subquery predicate in join condition is not supported");
+                Set<QualifiedName> dependencies = SymbolsExtractor.extractNames(complexExpression, analysis.getColumnReferences());
+
+                if (dependencies.stream().allMatch(left::canResolve)) {
+                    leftPlanBuilder = subqueryPlanner.handleSubqueries(leftPlanBuilder, complexExpression, node);
+                }
+                else {
```

Comment:
`else if (dependencies.stream().allMatch(right::canResolve)` and then `else fail`?

Reply:
It doesn't matter. This is for handling uncorrelated subqueries (correlated ones are checked and prevented during analysis, as they are unsupported). This is a best effort to plan the subquery in the branch of the join that involves the operand in the expression. E.g., 

* `SELECT ... FROM t LEFT JOIN u ON t.x IN (...)`, we want the subquery to be planned in the `t` branch
* `SELECT ... FROM t LEFT JOIN u ON u.x IN (...)`, we want the subquery to be planned in the `u` branch
* `SELECT ... FROM t LEFT JOIN u ON t.x + u.x IN (...)`, handle the subquery in an arbitrary branch. Not much we can do.

I'll add a comment with an explanation.

Reply:
thanks